daffodil-runtime1/src/main/scala/org/apache/daffodil/api/DFDLParserUnparser.scala

```diff
@@ -132,22 +140,57 @@ object DFDL {
      * To explicitly specify that there is no-namespace, pass "" as
      * the namespace argument.
      */
+    @deprecated("2020-03-26", "Use withDistinguishedRootNode.")
     def setDistinguishedRootNode(name: String, namespace: String = null)
+
+    def withDistinguishedRootNode(name: String, namespace: String) : ProcessorFactory
+
     def onPath(xpath: String): DataProcessor
   }
 
   trait DataProcessor extends WithDiagnostics {
-    def setValidationMode(mode: ValidationMode.Type): Unit
-    def getValidationMode(): ValidationMode.Type
+
+    /**
+     * Returns a data processor with the same schema, no external variable bindings,
+     * and the default validation mode.
+     */
+    def clear(): DataProcessor
```

Comment:
Is this just so users don't need to keep a reference to the original DataProcessor

Reply:
Yes. Also so that you can use the deprecated setters, and take the resulting DP and get a clean one to start from again. 

I'm not sure *we* need this method. I call it one place and I'm not sure it is needed there.